map-view/src/common/Map.ts

```diff
@@ -83,31 +91,124 @@ class Map {
+            feature["app_name"] = feature_layer ? feature_layer["app_name"] : "traffic_control";
+            return [feature];
+          }
         }
-        return features
+        return features;
       });
     }
 
     this.map.on("singleclick", (event) => {
-      const visibleLayers = Object.values(this.overlayLayers).filter((layer) => layer.getVisible());
+      const layers = { ...this.overlayLayers, extraVectorLayer: this.extraVectorLayer };
+      const visibleLayers = Object.values(layers).filter((layer) => layer.getVisible());
       if (visibleLayers.length > 0) {
         // Combine the topmost feature from all visible layers into a single array
         Promise.all(visibleLayers.map((layer) => getFeatureFromLayer(layer, event.pixel))).then((features) => {
+          // @ts-ignore
+          const all_features = [].concat.apply([], features);
+          if (all_features.length > 0) {
+            this.featureInfoCallback(all_features);
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Show a plan object of the selected real device on map
+   *
+   * @param feature Target feature of which the plan/real device will be shown
+   * @param mapConfig
+   */
+  showPlanOfRealDevice(feature: Feature, mapConfig: MapConfig) {
+    return new Promise((resolve) => {
+      if (feature.values_.device_plan_id) {
+        const { overlayConfig } = mapConfig;
+
+        // Find selected Real device's Plan layer's config
+        const featureType: string = feature["id_"].split(".")[0].replace("real", "plan");
+        const feature_layer = overlayConfig["layers"].find((l) => l.identifier === featureType);
+
+        if (feature_layer) {
+          const vectorSource = new VectorSource({
+            format: this.geojsonFormat,
+            url:
+              overlayConfig.sourceUrl +
+              `?SERVICE=WFS&VERSION=2.0.0&REQUEST=GetFeature&OUTPUTFORMAT=geojson&TYPENAMES=${feature_layer.identifier}
+&FILTER=%3CFilter%3E%3CResourceId%20rid=%22${feature_layer.identifier}.${feature.values_.device_plan_id}%22/%3E%3C/Filter%3E`,
+          });
+          this.extraVectorLayer.setSource(vectorSource);
+
+          // Add a line between the plan and real points
+          vectorSource.on("featuresloadend", (featureEvent) => {
+            const features = featureEvent.features;
+            if (features) {
+              // @ts-ignore
+              const plan_location = features[0].values_.geometry.flatCoordinates;
+              // @ts-ignore
+              const real_location = feature.values_.geometry.flatCoordinates;
+              const lineString = new LineString([plan_location, real_location]);
+              const olFeature = new OlFeature({
+                geometry: lineString,
+                name: "Line",
+              });
+              this.extraVectorLayer.getSource()!.addFeature(olFeature);
+
+              // Calculate distance between Real and Plan
+              const distance =
```

Comment:
This could be a separate function with some info about rounding the value to precision of two decimals